File: articles/organic-chemistry-carbon-based-structure/fortran/organic_structure_kernel.f90
program organic_structure_kernel
  implicit none

  real(8) :: benzene_dbe, acetic_acid_dbe, polarity_score

  benzene_dbe = 6.0d0 - (6.0d0 + 0.0d0) / 2.0d0 + 0.0d0 / 2.0d0 + 1.0d0
  acetic_acid_dbe = 2.0d0 - (4.0d0 + 0.0d0) / 2.0d0 + 0.0d0 / 2.0d0 + 1.0d0
  polarity_score = 2.0d0 + 1.0d0 + 2.0d0

  print *, "benzene_DBE=", benzene_dbe
  print *, "acetic_acid_DBE=", acetic_acid_dbe
  print *, "polarity_score=", polarity_score

end program organic_structure_kernel

File: articles/organic-chemistry-carbon-based-structure/julia/organic_structure_kernel.jl
# Organic-structure numerical kernel in Julia.

function dbe(C, H, N, X)
    return C - (H + X) / 2.0 + N / 2.0 + 1.0
end

function polarity_score(heteroatoms, donors, acceptors)
    return heteroatoms + donors + acceptors
end

function boltzmann_populations(energies_kj_mol, temperature_k)
    R = 0.008314462618
    weights = exp.(-energies_kj_mol ./ (R * temperature_k))
    return weights ./ sum(weights)
end

println("benzene_DBE=", round(dbe(6, 6, 0, 0), digits=6))
println("acetic_acid_DBE=", round(dbe(2, 4, 0, 0), digits=6))
println("polarity_score=", polarity_score(2, 1, 2))
println("conformer_populations=", round.(boltzmann_populations([0.0, 2.5, 6.0], 298.15), digits=6))
